### PR TITLE
PEP8 and lint code with flake8 on Travis CI

### PR DESCRIPTION
**Pull request by moyogo**
Should the unused variables and unused import from 5cf4904 and 0fa018b just be removed?

### CODE REVIEW

File: Lib/defconQt/__main__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from defconQt.objects.defcon import TFont
-from defconQt import icons_db, representationFactories
+from defconQt import representationFactories
```
Comment (adrientetar) [inline Lib/defconQt/__main__.py]: icons_db needs to stay, it packs the compiled images for the program.

File: Lib/defconQt/baseCodeEditor.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
-from PyQt5.QtCore import QRegularExpression, Qt
-from PyQt5.QtGui import QColor, QFont, QPainter, QSyntaxHighlighter, QTextCursor
+from PyQt5.QtCore import QRegularExpression, Qt, QSize
+from PyQt5.QtGui import (
+    QColor, QFont, QPainter, QSyntaxHighlighter, QTextCursor)
```
Comment (adrientetar) [inline Lib/defconQt/baseCodeEditor.py]: Why the ( ) and newline?

Comment (adrientetar) [inline Lib/defconQt/baseCodeEditor.py]: Okay it was too long of one line...

File: Lib/defconQt/fontView.py [hunk 3]
```diff
@@ -10,42 +10,58 @@
 from defcon import Color, Component
 from defconQt.spaceCenter import MainSpaceWindow
 # TODO: remove globs when things start to stabilize
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: You can remove this TODO now.

File: Lib/defconQt/fontView.py [hunk 4]
```diff
@@ -330,25 +361,31 @@ def writeUnicodes(self):
...
 
     def writeRightSideBearing(self):
-        if self._glyph is None: return
+        if self._glyph is None:
+            return
         self._blocked = True
         self._glyph.rightMargin = int(self.nameEdit.text())
         self._blocked = False
 
 # TODO: implement Frederik's Glyph Construction Builder
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: Is weird to have this comment in the middle of nowhere... maybe move it before the init function.

File: Lib/defconQt/fontView.py [hunk 5]
```diff
@@ -392,12 +430,12 @@ def __init__(self, currentGlyphs=None, parent=None):
     def getNewGlyphNames(cls, parent, currentGlyphs=None):
         dialog = cls(currentGlyphs, parent)
         result = dialog.exec_()
-        sortFont = False
+        # sortFont = False  # unused
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: Yes, this can be removed.

File: Lib/defconQt/fontView.py [hunk 6]
```diff
@@ -406,11 +444,13 @@ def getNewGlyphNames(cls, parent, currentGlyphs=None):
         if dialog.sortFontBox.isChecked():
             # XXX: if we get here with previous sort being by glyph set,
             # should it just stick?
-            sortFont = True
+            # sortFont = True  # unused
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: And here &ndash; the whole if condition can be removed.

File: Lib/defconQt/fontView.py [hunk 7]
```diff
@@ -590,7 +640,7 @@ def __init__(self, font):
         self.collectionWidget.setFocus()
 
         menuBar = self.menuBar()
-        settings = QSettings()
+        # settings = QSettings()  # unused
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: Can be removed.

File: Lib/defconQt/fontView.py [hunk 8]
```diff
@@ -901,31 +962,32 @@ def _set_sortDescriptor(self, desc):
...
-                .sortGlyphNames(self._font.keys(), desc)]
+                      .sortGlyphNames(self._font.keys(), desc)]
         self.collectionWidget.glyphs = glyphs
         self._sortDescriptor = desc
 
     sortDescriptor = property(_get_sortDescriptor, _set_sortDescriptor,
-        doc="The sortDescriptor. Takes glyphs from the font and sorts them \
-        when set.")
+                              doc="The sortDescriptor. Takes glyphs from the "
+                              "font and sorts them when set.")
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: Somehow making a new string like this concats to the previous one? Did not know.

File: Lib/defconQt/fontView.py [hunk 9]
```diff
@@ -901,31 +962,32 @@ def _set_sortDescriptor(self, desc):
...
 
     def getGlyphs(self):
         return self.collectionWidget.glyphs
 
     def copy(self):
         glyphs = self.collectionWidget.glyphs
-        selection = self.collectionWidget.selection
+        # selection = self.collectionWidget.selection  # unused
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: Can be removed.

File: Lib/defconQt/fontView.py [hunk 10]
```diff
@@ -901,31 +962,32 @@ def _set_sortDescriptor(self, desc):
...
         mimeData = QMimeData()
-        mimeData.setData("application/x-defconQt-glyph-data", pickle.dumps(pickled))
+        mimeData.setData("application/x-defconQt-glyph-data",
+                         pickle.dumps(pickled))
         clipboard.setMimeData(mimeData)
 
     def copyAsComponent(self):
         glyphs = self.collectionWidget.glyphs
-        selection = self.collectionWidget.selection
+        # selection = self.collectionWidget.selection  # unused
```
Comment (adrientetar) [inline Lib/defconQt/fontView.py]: Can be removed.

File: Lib/defconQt/glyphCollectionView.py [hunk 11]
```diff
@@ -14,8 +17,8 @@
 GlyphCellBufferHeight = .2
 GlyphCellHeaderHeight = 14
 
-# TODO: consider extracting each platform-specific thing (fonts, shortcuts) in a
-# purposed folder
+# TODO: consider extracting each platform-specific thing (fonts, shortcuts) in
+# a autpurposed folder
```
Comment (adrientetar) [inline Lib/defconQt/glyphCollectionView.py]: What does autpurposed mean?

Comment (adrientetar) [inline Lib/defconQt/glyphCollectionView.py]: By the way, this TODO can be removed now.

File: Lib/defconQt/glyphCollectionView.py [hunk 12]
```diff
@@ -229,31 +248,33 @@ def keyPressEvent(self, event):
...
             if proceedWithDeletion(self) and self.selection:
                 # we need to del in reverse order to keep key references valid
                 for key in sorted(self._selection, reverse=True):
                     glyph = self._glyphs[key]
-                    font = glyph.getParent()
+                    # font = glyph.getParent()  # unused
                     if modifiers & Qt.ShiftModifier:
-                        del self.font[gName]
+                        # del self.font[gName]
+                        # # gName is undefined
```
Comment (adrientetar) [inline Lib/defconQt/glyphCollectionView.py]: Keep the font variable up here and replace self.font &rarr; font and gName &rarr; glyph.name.

File: Lib/defconQt/glyphCollectionView.py [hunk 13]
```diff
@@ -375,16 +404,18 @@ def mouseMoveEvent(self, event):
...
-                glyphList = " ".join(glyph.name for glyph in self.getSelectedGlyphs())
+                glyphList = " ".join(
+                    glyph.name for glyph in self.getSelectedGlyphs())
                 drag = QDrag(self)
                 mimeData = QMimeData()
                 mimeData.setText(glyphList)
                 drag.setMimeData(mimeData)
 
-                dropAction = drag.exec_()
+                # dropAction = drag.exec_()  # unused
```
Comment (adrientetar) [inline Lib/defconQt/glyphCollectionView.py]: drag.exec_() needs to stay in, only the result code is not used and so dropAction variable can be removed.

File: Lib/defconQt/glyphCollectionView.py [hunk 14]
```diff
@@ -553,9 +610,13 @@ def paintEvent(self, event):
...
-                    rect = QRectF(column * self.squareSize, row * self.squareSize+GlyphCellHeaderHeight,
-                          self.squareSize, self.squareSize-GlyphCellHeaderHeight)
-                    # TODO: need to flag template glyphs as to whether they have unicodings or not
+                    rect = QRectF(
+                        column * self.squareSize,
+                        row * self.squareSize + GlyphCellHeaderHeight,
+                        self.squareSize,
+                        self.squareSize - GlyphCellHeaderHeight)
+                    # TODO: need to flag template glyphs as to whether they
+                    # have unicodings or not
```
Comment (adrientetar) [inline Lib/defconQt/glyphCollectionView.py]: This comment is outdated and can be cleaned up.

File: Lib/defconQt/layerSetList.py [hunk 15]
```diff
@@ -122,8 +125,10 @@ def _changeColor(self):
         if not layer:
             return
 
-        startColor = layer.color and QColor.fromRgbF(*layer.color) or QColor('limegreen')
-        qcolor = QColorDialog.getColor(startColor, self, options=QColorDialog.ShowAlphaChannel)
+        startColor = layer.color and QColor.fromRgbF(*layer.color) or \
+            QColor('limegreen')
+        qcolor = QColorDialog.getColor(
```
Comment (adrientetar) [inline Lib/defconQt/layerSetList.py]: This should be named newColor.

File: Lib/defconQt/pens/copySelectionPen.py [hunk 16]
```diff
@@ -40,7 +44,7 @@ def addPoint(self, pt, segmentType=None, smooth=False, name=None, selected=False
                 self.beginPath()
 
     def addComponent(self, baseGlyphName, transformation):
-        pass # XXX
+        pass  # XXX
```
Comment (adrientetar) [inline Lib/defconQt/pens/copySelectionPen.py]: Weird.

File: Lib/defconQt/spaceCenter.py [hunk 17]
```diff
@@ -1,19 +1,24 @@
-from defconQt import icons_db
+# from defconQt import icons_db  # unused
```
Comment (adrientetar) [inline Lib/defconQt/spaceCenter.py]: This needs to stay as well.

File: Lib/defconQt/spaceCenter.py [hunk 18]
```diff
@@ -470,34 +509,38 @@ def mousePressEvent(self, event):
...
+                self._selected = None
             if self.selectionChangedCallback is not None:
                 self.selectionChangedCallback(self._selected)
             event.accept()
             self.update()
-        else:
-            super(GlyphCanvas, self).mousePressEvent(event)
+        # else:
+        #     super(GlyphCanvas, self).mousePressEvent(event)
+        # # GlyphCanvas is undefined
```
Comment (adrientetar) [inline Lib/defconQt/spaceCenter.py]: Replace GlyphCanvas &rarr; GlyphsCanvas and uncomment here.

File: Lib/defconQt/spaceCenter.py [hunk 19]
```diff
@@ -470,34 +509,38 @@ def mousePressEvent(self, event):
...
 
     def mouseDoubleClickEvent(self, event):
         if event.button() == Qt.LeftButton and self._selected is not None:
             if self.doubleClickCallback is not None:
                 self.doubleClickCallback(self.glyphs[self._selected])
-        else:
-            super(GlyphCanvas, self).mouseDoubleClickEvent(event)
+        # else:
+        #     super(GlyphCanvas, self).mouseDoubleClickEvent(event)
+        # # GlyphCanvas is undefined
```
Comment (adrientetar) [inline Lib/defconQt/spaceCenter.py]: Same here.

File: setup.cfg [hunk 20]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+# ignore = E226
+max-line-length = 79
+# max-complexity should be 10
```
Comment (adrientetar) [inline setup.cfg]: haha

File: Lib/defconQt/spaceCenter.py [hunk 21]
```diff
@@ -450,12 +488,13 @@ def mousePressEvent(self, event):
...
         self.setFocus(Qt.MouseFocusReason)
         if event.button() == Qt.LeftButton:
-            if self._verticalFlip:
-                baselineShift = -self.descender
-            else:
-                baselineShift = self.ascender
+            # if self._verticalFlip:
+            #     baselineShift = -self.descender
+            # else:
+            #     baselineShift = self.ascender  # unused
```
Comment (adrientetar) [inline Lib/defconQt/spaceCenter.py]: Remove the unused comment and put a FIXME on top of the comment block that this should be investigated.

**Comment by adrientetar** [comment]
Okay I have disabled landscape.io because it does not support Python 3, w my comments above this will be ready to merge &ndash; thanks!

**Comment by moyogo** [comment]
OK.
I've also uncommented `lineObj = HandleLineItem(...)` when lineObj is unused since the HandleLineItem.__init__() is needed and added a comment like https://github.com/moyogo/trufont/blob/flake8/Lib/defconQt/glyphView.py#L1463

**Comment by adrientetar** [comment]
So just `HandleLineItem(0, 0, 0, 0, item)` does not instanciate the object? Okay. There is no need for a XXX in this case because the item is not meant to be used whatsoever.

**Comment by adrientetar** [comment]
Just enabled Travis, thanks!

**Comment by moyogo** [comment]
Thanks!

**Pull request merged**